### PR TITLE
Updating gradient clipping to be torch 2.0 compatible

### PR DESCRIPTION
**Pull request by bcui19**
# What does this PR do?
In torch 2.0 they change the error message that is thrown for gradient clipping. This PR adds code to catch this new error message.

ran locally with torch 2.0:
https://wandb.ai/mosaic-ml/examples_fork-examples_llm/runs/prr9w4gb?workspace=user-bcui

link to PyTorch 2.0 new error message:
https://github.com/pytorch/pytorch/blob/master/torch/distributed/fsdp/fully_sharded_data_parallel.py#L1000

# What issue(s) does this change relate to?
[CO-1955](https://mosaicml.atlassian.net/browse/CO-1955)

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

[CO-1955]: https://mosaicml.atlassian.net/browse/CO-1955?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: composer/algorithms/gradient_clipping/gradient_clipping.py [hunk 1]
```diff
@@ -42,6 +42,11 @@ def apply_gradient_clipping(model: Union[ComposerModel, torch.nn.Module], clippi
         if version.parse(torch.__version__) < version.parse('1.13.0'):
             raise RuntimeError('To use FSDP with Composer, you must use torch>=1.13.0.')
         from torch.distributed.fsdp import FullyShardedDataParallel
+
```
Comment (dskhudia) [inline composer/algorithms/gradient_clipping/gradient_clipping.py:45]: I suspect this pattern is used at multiple places already and will be used at more places. Should we create a function for this that caches this result?

File: composer/algorithms/gradient_clipping/gradient_clipping.py [hunk 2]
```diff
@@ -55,8 +60,13 @@ def apply_gradient_clipping(model: Union[ComposerModel, torch.nn.Module], clippi
                     else:
                         raise ValueError(f"clipping type must be 'norm' or 'value' with FSDP not {clipping_type}")
                 except AssertionError as e:
-                    # Catches the error message from PyTorch
-                    if 'clip_grad_norm should only be called on the root (parent) instance' == str(e):
+                    # Catches the error message from PyTorch 1.13
+                    if 'clip_grad_norm should only be called on the root (parent) instance' == str(
```
Comment (dskhudia) [inline composer/algorithms/gradient_clipping/gradient_clipping.py:64]: Instead of doing exception catching twice, do you want to add an OR here?
